pkg/cli-runtime/testing/client.go

```diff
@@ -50,7 +59,132 @@ func NewFakeCliClient(c crclient.Client) cli.Client {
 	}
 }
 
+// RESTClient returns a REST client from TestFactory
+func (c *fakeclient) RESTClient() (*rest.RESTClient, error) {
+	// Swap out the HTTP client out of the client with the fake's version.
+	fakeClient := c.Clients.(*fake.RESTClient)
+	restClient, err := rest.RESTClientFor(c.ClientConfigVal)
+	if err != nil {
+		panic(err)
+	}
+	restClient.Client = fakeClient.Client
+	return restClient, nil
+}
+func (c *fakeclient) RESTMapper() meta.RESTMapper {
+	return testRESTMapper()
+}
+func (c *fakeclient) ToRESTMapper() (meta.RESTMapper, error) {
+	return c.RESTMapper(), nil
+}
+func (c *fakeclient) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
+	return disk.NewCachedDiscoveryClientForConfig(&rest.Config{}, "", "", 0) // need alignment with sash
+}
+func (c *fakeclient) ToRESTConfig() (*rest.Config, error) {
+	return c.KubeRestConfig(), nil
+}
+func NewFakeCachedDiscoveryClient() *FakeCachedDiscoveryClient {
+	return &FakeCachedDiscoveryClient{
+		Groups:             []*metav1.APIGroup{},
+		Resources:          []*metav1.APIResourceList{},
+		PreferredResources: []*metav1.APIResourceList{},
+		Invalidations:      0,
+	}
+}
+func (d *FakeCachedDiscoveryClient) ServerGroupsAndResources() ([]*metav1.APIGroup, []*metav1.APIResourceList, error) {
+	return d.Groups, d.Resources, nil
+}
+
+// NewBuilder returns a new resource builder for structured api objects.
+func (c *fakeclient) NewBuilder() *resource.Builder {
+	return resource.NewFakeBuilder(
```

Comment:
no test is accessing fake builder from this func so why do you need this? return `panic(not implemented)` for this func